compiler/dialects/accfg.md

```diff
@@ -0,0 +1,141 @@
+# `accfg` Dialect Documentation
+
+The `accfg` dialect is meant to server as a genric optimisation dialect, which may be used by any lowering pipieline
+who has certain semantics and wants to use our optimisations. It is therefore paramount that the dialect documents
+the semantics of its operations and the assumptions made by its passes.
+
+## Operation Semantics
+
+The `accfg` dialect provides three operations that interact with a stateful "client" device. These operations are as
```

Comment:
I find the use of the word `client` in this context a bit confusing. I think we better keep it very concrete here for now, if other people find other use-cases for this, we can change this wording later, but now we don't have other use-cases, so I'd just keep it like this?
```suggestion
The `accfg` dialect provides three operations that interact with a stateful accelerator device. These operations are as
```

Reply:
I think `accfg.accelerator` is more an artifact of MLIR not having target information (which is about to change), rather than an integral part of the dialect.